modern/src/App.js

```diff
@@ -302,10 +352,8 @@ function XmlNode({ node }) {
     return null;
   }
   return (
-    <GuiObjectEvents node={node}>
-      <Component node={node} {...attributes}>
-        {children}
-      </Component>
+    <GuiObjectEvents Component={Component} node={node}>
+      {children}
```

Comment:
We needed to move the `Component` creation into `GuiObjectEvents` to get updating to work properly